src/VisualLinter/Linting/Linter.cs

```diff
@@ -37,34 +37,74 @@ internal class Linter : ILinter
                     var directoryPath = Path.GetDirectoryName(filePath) ??
                         throw new Exception($"exception: could not get directory for file {filePath}");
 
-                    var eslintPath = EslintHelper.GetEslintPath(directoryPath);
-                    var arguments = string.Join(" ", QuoteArgument(filePath), EslintHelper.GetArguments(directoryPath));
+                    LinterResults linterResults = new LinterResults();
 
-                    var output = await RunAsync(eslintPath, arguments, token)
-                        .ConfigureAwait(false);
+                    string fileExtension = Path.GetExtension(filePath).ToLower();
 
-                    token.ThrowIfCancellationRequested();
+                    if (fileExtension == ".html" || fileExtension == ".js" || fileExtension == ".jsx" || fileExtension == ".vue")
+                    {
+                        var eslintPath = EslintHelper.GetEslintPath(directoryPath);
+                        var arguments = string.Join(" ", QuoteArgument(filePath), EslintHelper.GetArguments(directoryPath));
 
-                    if (string.IsNullOrEmpty(output))
-                        throw new Exception("exception: linter returned empty result");
+                        var output = await RunAsync("eslint", eslintPath, arguments, token)
+                            .ConfigureAwait(false);
 
-                    IEnumerable<EslintResult> results = new List<EslintResult>();
+                        token.ThrowIfCancellationRequested();
 
-                    try
-                    {
-                        results = JsonConvert.DeserializeObject<IEnumerable<EslintResult>>(output);
+                        if (string.IsNullOrEmpty(output))
+                            throw new Exception("exception: eslint returned empty result");
+
+                        EslintResults eslintResults = new EslintResults();
+
+                        try
+                        {
+                            eslintResults = JsonConvert.DeserializeObject<EslintResults>(output);
+                        }
+                        catch (Exception e)
+                        {
+                            OutputWindowHelper.WriteLine(
+                                "exception: error trying to deserialize output:" +
+                                Environment.NewLine +
+                                output);
+
+                            OutputWindowHelper.WriteLine(e.Message);
+                        }
+
+                        linterResults = new LinterResults(eslintResults);
                     }
-                    catch (Exception e)
+                    else if (fileExtension == ".css" || fileExtension == ".scss" || fileExtension == ".sass" || fileExtension == ".less")
```

Comment:
There's way to much copy and paste in the `Linter.cs` class by now, and respective logic should be moved into `EslintLinter.cs` (for example) and `StylelintLinter.cs` respectively. 

`Linter.cs` class should, instead of having all the `if`s introduce a new method called `GetLinter(string extension)` and then call `RunAsync` on that specific linter (either `EslintLinter` or `StylelintLinter`.